Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -875,44 +875,37 @@ double TimeSeriesProperty<std::string>::averageValueInFilter(const SplittingInte
 
 template <typename TYPE>
 std::pair<double, double>
-TimeSeriesProperty<TYPE>::averageAndStdDevInFilter(const std::vector<SplittingInterval> &filter) const {
-  // the mean to calculate the standard deviation about
-  // this will sort the log as necessary as well
-  const double mean = this->averageValueInFilter(filter);
+TimeSeriesProperty<TYPE>::averageAndStdDevInFilter(const std::vector<SplittingInterval> &intervals) const {
+  double mean_prev, mean_current(0.0), s(0.0), variance, duration, weighted_sum(0.0);
 
-  // First of all, if the log or the filter is empty or is a single value,
-  // return NaN for the uncertainty
-  if (realSize() <= 1 || filter.empty()) {
-    return std::pair<double, double>{mean, std::numeric_limits<double>::quiet_NaN()};
+  if (realSize() <= 1 || intervals.empty()) {
+    return std::pair<double, double>{this->averageValueInFilter(intervals), std::numeric_limits<double>::quiet_NaN()};
   }
 
-  double numerator(0.0), totalTime(0.0);
-  // Loop through the filter ranges
-  for (const auto &time : filter) {
-    // Calculate the total time duration (in seconds) within by the filter
-    totalTime += time.duration();
-
-    // Get the log value and index at the start time of the filter
+  for (const auto &time : intervals) {
     int index;
-    double value = static_cast<double>(getSingleValue(time.begin(), index));
-    double valuestddev = (value - mean) * (value - mean);
+    auto value = static_cast<double>(getSingleValue(time.begin(), index));
     DateAndTime startTime = time.begin();
+    while (index < realSize()) {
```

Comment:
I don't know if it'll have a big performance effect, but I'd feel better if `realSize()` were cached outside of the `for` loop that starts on L885

Reply:
Latest commit reflects these changes.